File: src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/EventHeaderDTO.java
package uk.gov.ons.ssdc.caseprocessor.model.dto;

import java.time.OffsetDateTime;
import java.util.UUID;
import lombok.Data;

@Data
public class EventHeaderDTO {
  private String topic;
  private String source;
  private String channel;
  private OffsetDateTime dateTime;
  private UUID messageId;
  private UUID correlationId;
  private String originatingUser;
}

File: python_spike/caseprocessor/dto/event_header_dto.py
import uuid
from dataclasses import dataclass, field
from datetime import datetime
from dataclasses_json import LetterCase, dataclass_json, config
from marshmallow import fields
from typing import Optional


@dataclass_json(letter_case=LetterCase.CAMEL)
@dataclass
class EventHeaderDTO:
    topic: str
    source: str
    channel: str
    correlation_id: uuid
    originating_user: str
    date_time: Optional[datetime] = field(
        default=None,
        metadata=config(
            encoder=lambda time: datetime.isoformat(time) if time else None,
            decoder=lambda time: datetime.fromisoformat(time) if time else None,
            mm_field=fields.DateTime(format='iso')
        )
    )
    message_id: uuid = None
    version: str = None
